tests/nodeos_extra_packed_data_test.py

```diff
@@ -0,0 +1,199 @@
+    testerbAccount.name="testerxxxxxb"
+
+    testWalletName="test"
+    Print("Creating wallet \"%s\"" % (testWalletName))
+    walletAccounts=copy.deepcopy(cluster.defProducerAccounts)
+    if dontLaunch:
+        del walletAccounts["eosio"]
+    testWallet = walletMgr.create(testWalletName, walletAccounts.values())
+
+    Print("Wallet \"%s\" password=%s." % (testWalletName, testWallet.password.encode("utf-8")))
+
+    all_acc = accounts + list( cluster.defProducerAccounts.values() )
+    for account in all_acc:
+        Print("Importing keys for account %s into wallet %s." % (account.name, testWallet.name))
+        if not walletMgr.importKey(account, testWallet):
+            cmdError("%s wallet import" % (ClientName))
+            errorExit("Failed to import key for account %s" % (account.name))
+    
+    node=cluster.getNode(0)
+
+    Print("Create new account %s via %s" % (testeraAccount.name, cluster.defproduceraAccount.name))
+    transId=node.createInitializeAccount(testeraAccount, cluster.defproduceraAccount, stakedDeposit=0, waitForTransBlock=False, exitOnError=True)
+
+    Print("Create new account %s via %s" % (testerbAccount.name, cluster.defproduceraAccount.name))
+    transId=node.createInitializeAccount(testerbAccount, cluster.defproduceraAccount, stakedDeposit=0, waitForTransBlock=False, exitOnError=True)
+
+    Print("Validating accounts after user accounts creation")
+    accounts=[testeraAccount, testerbAccount]
+    cluster.validateAccounts(accounts)
+
+    for i in range(2):
+        if i == 1:
+            node = cluster.getNode(prodCount - 1)
+        
+        transferAmount="{0}.0 {1}".format(i + 1, CORE_SYMBOL)
+        Print("Transfer funds %s from account %s to %s" % (transferAmount, cluster.defproduceraAccount.name, testeraAccount.name))
+        trx = node.transferFunds(cluster.defproduceraAccount, testeraAccount, transferAmount, "test transfer", dontSend=True)
+
+        cmdDesc = "convert pack_transaction"
+        cmd     = "%s --pack-action-data '%s'" % (cmdDesc, json.dumps(trx))
+        exitMsg = "failed to pack transaction: %s" % (trx)
+        packedTrx = node.processCleosCmd(cmd, cmdDesc, silentErrors=False, exitOnError=True, exitMsg=exitMsg)
+
+        packed_trx_param = packedTrx["packed_trx"]
+        if packed_trx_param is None:
+            cmdError("packed_trx is None. Json: %s" % (packedTrx))
+            errorExit("Can't find packed_trx in packed json")
+        
+        #adding random trailing data
+        packedTrx["packed_trx"] = packed_trx_param + "00000000"
+
+        exitMsg = "failed to send packed transaction: %s" % (packedTrx)
+        node.processCurlCmd("chain", "send_transaction", json.dumps(packedTrx), silentErrors=False, exitOnError=True, exitMsg=exitMsg)
+
+    info = node.getInfo(exitOnError=True)
+    oldLIB = int(info["last_irreversible_block_num"])
+
+    blocksGap = (12 * (2 * math.floor(2 * prodCount / 3)) + 12) * 1.3 #adding 30% more to be on the safe side
```

Comment:
I'm not following exactly what this remaining section is testing for, but I think we could greatly reduce the time this takes.  I think we should look for the transactions in blocks (using getTransaction(..finalized)), for both node(0) and node(prod-1), and verify that node(0) and node(prod-1) both have advanced LIB past oldLIB and that head has progressed at least 12*pnode blocks, if you do node.getBlock(lastTransactionBlock+12*pnode) it will do the waiting if you need it to, just make sure you check that it returns a block (getBlock unfortunately just returns None on timeout, so need to verify)

Reply:
I fixed everything you suggested but I don't follow this part. so we need not just block + 12. We need LIB + 12.
`cleos get block` returns any block, including irreversible. I checked the code. So how we can wait till LIB move more efficiently?

Reply:
that was supposed to be:
`node.getBlock(lastTransactionBlock+12 * pnode)`

Reply:
but it doesn't return LIB, isn't it?

Reply:
basically, waiting long enough to verify that all producers built off of those blocks.  Of course it doesn't prove it, so, you could also retrieve each of those blocks and put the block["producer"] in a dictionary and verify that len(prodDict) == pnode, That would be a thorough test.  You would only need to do it till while ( len(prodDict) < pnode), and then fail if you got past
`lastTransactionBlock + 12 * pnode`

Reply:
@brianjohnson5972 ok. original test was related to LIB being stuck so I made waiting till LIB move. If your mentioned test is enough and we don't need to wait for LIB (if stuck block in node reproduces it) than I'm ok to make those changes. please confirm.

Reply:
"original test" meaning the test you copied it from?  If yes, then I think this is sufficient, since we will be verifying that each producer was able to process it.